File: questions/billPay/billpay.js
const { expect } = require("chai");
const Mocha = require("mocha");
const mocha = new Mocha();
mocha.suite.emit("pre-require", this, "solution", mocha); // Bit of a hack, sorry!


class Bill {

  constructor(date) {
    this.date = date;
  }

  get paymentDate() {
    return null;
  }
}


const TestingDates = {
  wednesday: new Date("01/02/2019"),
  thursday: new Date("01/03/2019"),
  friday: new Date("01/04/2019"),
  saturday: new Date("01/05/2019"),
  sunday: new Date("01/06/2019"),
  nextMonday: new Date("01/07/2019"),
  nextTuesday: new Date("01/08/2019")
};


describe("Bill", () => {
  describe("paymentDate", () => {
    it("Should return the payment date", () => {
      expect(new Bill(TestingDates.wednesday).paymentDate.toString()).to.equal(TestingDates.wednesday.toString());
    });
  })
});


mocha.run();

File: questions/billPay/billpay.py
from datetime import datetime
from unittest import TestCase, main as unitTestMain


class Bill():
    def __init__(self, date):
        self.date = date

    def get_payment_date(self):
        raise NotImplementedError()


class TestingDates:
    DATE_FORMAT = '%m/%d/%y'

    THURSDAY = datetime.strptime('10/18/18', DATE_FORMAT)
    FRIDAY = datetime.strptime('10/19/18', DATE_FORMAT)
    SATURDAY = datetime.strptime('10/20/18', DATE_FORMAT)
    SUNDAY = datetime.strptime('10/21/18', DATE_FORMAT)
    NEXT_MONDAY = datetime.strptime('10/22/18', DATE_FORMAT)
    NEXT_TUESDAY = datetime.strptime('10/23/18', DATE_FORMAT)
    NEXT_WEDNESDAY = datetime.strptime('10/24/18', DATE_FORMAT)


class TestBill(TestCase):

    def test_returns_payment_date(self):
        payment_date = Bill(TestingDates.THURSDAY).get_payment_date()
        self.assertEqual(TestingDates.THURSDAY, payment_date)


if __name__ == '__main__':
    unitTestMain()
